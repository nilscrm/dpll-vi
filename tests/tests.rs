use insta_cmd::assert_cmd_snapshot;
use rstest::rstest;
use std::{fs::File, path::{Path, PathBuf}, process::Command};

fn test_cnf_files(group_name: &str, path: &Path) {
  insta::with_settings!({
    snapshot_path => format!("snapshots/{}", group_name),
    filters => [(r"Performance\s+Time\s+.*\sms\s+Speed.*IPS", "[Performance...]")] },
  {
    let stdin = File::open(path).unwrap();
    assert_cmd_snapshot!(path.file_stem().unwrap().to_str().unwrap(), Command::new("vine").args(["run", "dpll/main.vi"]).arg("--breadth-first").stdin(stdin));
  });
}

#[rstest]
fn test_minimal(#[files("tests/minimal/*.cnf")] path: PathBuf) {
  test_cnf_files("minimal", &path);
}

#[rstest]
fn test_medium(#[files("tests/medium/*.cnf")] path: PathBuf) {
  test_cnf_files("medium", &path);
}

#[rstest]
fn test_satlib_sat(#[files("tests/uf5-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uf5-218_first_10", &path);
}

#[rstest]
fn test_satlib_unsat(#[files("tests/uuf50-218_first_10/*.cnf")] path: PathBuf) {
  test_cnf_files("uuf50-218_first_10", &path);
}
