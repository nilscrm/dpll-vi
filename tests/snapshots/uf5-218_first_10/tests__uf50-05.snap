---
source: tests/tests.rs
info:
  program: vine
  args:
    - run
    - dpll/main.vi
    - "--breadth-first"
---
success: true
exit_code: 0
----- stdout -----
satisfiable
evaluation under model: true
var 1 = true
var 2 = false
var 3 = true
var 4 = false
var 5 = true
var 6 = true
var 7 = false
var 8 = true
var 9 = false
var 10 = true
var 11 = false
var 12 = false
var 13 = false
var 14 = true
var 15 = true
var 16 = true
var 17 = true
var 18 = true
var 19 = true
var 20 = true
var 21 = true
var 22 = false
var 23 = true
var 24 = false
var 25 = true
var 26 = true
var 27 = false
var 28 = true
var 30 = true
var 31 = true
var 32 = true
var 33 = true
var 34 = false
var 35 = true
var 36 = true
var 37 = true
var 38 = false
var 39 = false
var 40 = true
var 41 = false
var 42 = true
var 43 = false
var 44 = true
var 45 = true
var 46 = true
var 47 = false
var 48 = false
var 49 = true
var 50 = true

----- stderr -----

Interactions
  Total            6_429_227
    Depth          3_182_634
    Breadth                2
  Annihilate       3_589_092
  Commute             23_613
  Copy               659_387
  Erase              741_406
  Expand             495_025
  Call               586_443
  Branch             334_261

Memory
  Heap               512_704 B
  Allocated      142_516_928 B
  Freed          142_516_928 B

[Performance...]
